src/app/components/shared/formly/image-input.component.spec.ts

```diff
@@ -1 +1,78 @@
-// TODO
+import {
+    FormControl,
+    FormGroup,
+    FormsModule,
+    ReactiveFormsModule,
+  } from "@angular/forms";
+import { createHostFactory, SpectatorHost } from "@ngneat/spectator";
+import { FormlyBootstrapModule } from "@ngx-formly/bootstrap";
+import { FormlyFieldProps, FormlyModule } from "@ngx-formly/core";
+import { ImageInputComponent } from "./image-input.component";
+import { formlyConfig } from "./custom-inputs.module";
+
+describe("FormlyImageInput", () => {
+    let model: any;
+    let formGroup: FormGroup;
+    let spectator: SpectatorHost<ImageInputComponent>;
+    const createHost = createHostFactory({
+      component: ImageInputComponent,
+      imports: [
+        FormsModule,
+        ReactiveFormsModule,
+        FormlyModule.forRoot(formlyConfig),
+        FormlyBootstrapModule,
+      ],
+    });
+    function getInput() {
+      return spectator.query<HTMLInputElement>("input[type='file']");
+    }
+
+    function getButton() {
+        return spectator.query<HTMLButtonElement>("button");
+    }
+
+    function setup(key: string = "file", options: FormlyFieldProps = {}) {
+      formGroup = new FormGroup({ asFormControl: new FormControl("") });
+      model = {};
+
+      spectator = createHost(
+        `
+        <form [formGroup]="formGroup">
+          <baw-image-input></baw-image-input>
+        </form>
+        `,
+        {
+          hostProps: { formGroup },
+          props: {
+            field: {
+              model,
+              key,
+              formControl: formGroup.get("asFormControl"),
+              props: options,
+            },
+          },
+        }
+      );
+      spectator.detectChanges();
+    }
+
+    describe("imageInput", () => {
+        it("should display file input", () => {
+            setup();
+            expect(getInput()).toBeTruthy();
+        });
+    });
+
+    describe("remove image", () => {
+        it("should set model value to null on click", () => {
+            setup();
+            getButton().click();
+            expect(formGroup.get("asFormControl").value).toBeNull();
+        });
```

Comment:
[10:15] Anthony Truskinger
if you were testing an attribute on model is set to null then i'd expect something along the lines of

expect(model.image).toBeNull();